test/numeric_test.ml

```diff
@@ -0,0 +1,142 @@
+        \n\226\148\130 tiny     \226\148\130 small     \226\148\130 int     \226\148\130 big     \226\148\130 float \226\148\130 double \226\148\130\
+        \n\226\148\130 Tiny_int \226\148\130 Small_int \226\148\130 Integer \226\148\130 Big_int \226\148\130 Float \226\148\130 Double \226\148\130\
+        \n\226\148\156\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\188\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\188\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\188\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\188\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\188\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\164\
+        \n\226\148\130 42       \226\148\130 42        \226\148\130 42      \226\148\130 42      \226\148\130 42.   \226\148\130 42.    \226\148\130\
+        \n\226\148\148\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\180\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\180\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\180\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\180\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\180\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\128\226\148\152\
+        \n"
+        |}]))
+;;
+
+let%expect_test "numeric_functions" =
+  Duckdb.Database.with_path ":memory:" ~f:(fun db ->
+    Duckdb.Connection.with_connection db ~f:(fun conn ->
+      (* Set DuckDB to single-threaded mode to avoid thread safety issues *)
+      Duckdb.Query.run_exn' conn "SET threads TO 1";
+      (* Test numeric functions *)
+      Duckdb.Query.run_exn
+        conn
+        {|
+        SELECT 
+          ABS(-42) as abs_val,
+          ROUND(42.4242, 2) as round_val,
+          FLOOR(42.9) as floor_val,
+          CEILING(42.1) as ceil_val
+        |}
+        ~f:(fun res ->
+          Duckdb.Result_.to_string_hum res ~bars:`Unicode |> [%sexp_of: string] |> print_s);
+      [%expect.unreachable]))
+[@@expect.uncaught_exn {|
```

Comment:
Unfortunately duckdb-ocaml doesn't have good support for decimals yet; let's wrap this exception in an Expect_test_helper_core call and add a TODO comment reminding ourselves to add support for it.